### PR TITLE
CLNRest: performance improvements using sql query

### PR DESCRIPTION
**Pull request by niteshbalusu11**
# Description

Relates to issue: #1637

We spoke about this a bit on the telegram chat, to begin with we will limit the list of transactions to 150. Right now I was able to limit the expensive `bkpr-listaccountevents` and `listinvoices`. I couldn't restrict `listpays` because it doesn't take a limit argument. There is a `listsendpays` that can take a limit but it doesn't return bolt12 payments (classic CLN). 

This pull request is categorized as a:

- [ ] New feature
- [ ] Bug fix
- [X] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance
- [ ] Other

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [X] Android
- [ ] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] Embedded LND
- [ ] LND (REST)
- [ ] LND (Lightning Node Connect)
- [X] Core Lightning (CLNRest)
- [ ] Core Lightning (c-lightning-REST)
- [ ] LndHub
- [ ] [DEPRECATED] Core Lightning (Spark)
- [ ] [DEPRECATED] Eclair

### CODE REVIEW

File: backends/CLNRest.ts [hunk 1]
```diff
@@ -210,7 +210,8 @@ export default class CLNRest {
         return this.postRequest('/v1/withdraw', request);
     };
     getMyNodeInfo = () => this.postRequest('/v1/getinfo');
-    getInvoices = () => this.postRequest('/v1/listinvoices');
+    getInvoices = () =>
+        this.postRequest('/v1/listinvoices', { limit: 150, index: 'created' });
```
Comment (Sjors) [inline backends/CLNRest.ts:214]: This seems like it might help with my 10k+ transactions :-)

(could make this paginate in a followup)

Comment (niteshbalusu11) [reply backends/CLNRest.ts:214]: Yeah, that's the idea.

**Comment by newtonick** [comment]
I did some testing on this PR. I checked out the PR locally and built from source. I replaced this `/v1/listinvoices` call with the SQL mentioned above. I did this so the output on the activity page was identical (at least for the visible screen).

It's a noticeable performance increase. I'm running my node on the same local network my phone is connecting to via WIFI. I'm comparing to the current v0.9.0-beta1 release from yesterday to this PR built from source. The activity view loads around 6-7 seconds faster.

Here is a screen recording of the performance. The first (light green) activity view loaded is v0.9.0-beta1 (not this PR). The second (yellow) activity view loaded is this PR. I have lurker privacy mode turned on for the recording.

https://github.com/user-attachments/assets/6f808d02-b3eb-4e33-a37e-56728eea935e

**Comment by niteshbalusu11** [comment]
Nice. Thank you for testing this. This again isn't the perfect solution because of the 150 records limitation but at least it makes the activity view usable.

**Comment by kaloudis** [comment]
Let's try this in `v09.0-beta2`

**Review by kaloudis — APPROVED** [review]

**Comment by niteshbalusu11** [comment]
@Sjors can you test RC2 please?

**Comment by Sjors** [comment]
Just tested, I can see my transactions now! 🎉 

When I click on a (sent) lightning transaction it immediately crashes.

Clicking on on chain does behave.

I get the impression that I'm not seeing _received_ lightning transactions, though would be mostly zaps and keysend from podcast streaming.

**Comment by Sjors** [comment]
By the way, I'm not a huge fan of the "€1,00 + €0,01 fee" notation. Especially for lightning fees are usually irrelevant and it just seems like visual clutter. It would be better to display fees only if you click on a transaction to see details.

What I _would_ like to see in the overview is a description. Maybe that can go where it says "lightning". And then "You sent" could say "You sent (⚡)"

**Comment by niteshbalusu11** [comment]
> Just tested, I can see my transactions now! 🎉 
> 
> When I click on a (sent) lightning transaction it immediately crashes.
> 
> Clicking on on chain does behave.
> 
> I get the impression that I'm not seeing _received_ lightning transactions, though would be mostly zaps and keysend from podcast streaming.

Will look into this.

**Comment by Sjors** [comment]
As of CLN v24.11 the `listpays` command has `index`, `start` and `limit` arguments. Might be good to leverage those.

**Pull request merged**